server/src/main/java/org/elasticsearch/cluster/metadata/MetadataUpgradeService.java

```diff
@@ -93,6 +101,42 @@ public IndexMetadata upgradeIndexMetadata(IndexMetadata indexMetadata,
         return markAsUpgraded(newMetadata);
     }
 
+    public Metadata upgradeRelationMetadata(Metadata metadata) {
+        Metadata.Builder newMetadata = Metadata.builder(metadata);
+        for (IndexMetadata indexMetadata : metadata) {
+            String indexName = indexMetadata.getIndex().getName();
+            if (IndexName.isPartitioned(indexName) == false) {
```

Comment:
What about partitioned tables? It should also check templates

Reply:
added and pushed, they are tested through integ tests. Should I also add some unit test logic in `GatewayMetaStateTests` as well?